### PR TITLE
feat: log http POST requests body in http-access log

### PR DESCRIPTION
**Pull request by stbrody**
Example log line from the POST request to create a new stream with the genesis commit:
```
[Tue, 23 Apr 2024 21:11:40 GMT] service=http-access ip=127.0.0.1 ts=2024-04-23T21:11:40.348Z method=POST original_url=/api/v0/streams base_url=/api/v0/streams path=/ params=-  body.type=0 body.genesis={"jws":{"payload":"AXESIDoayVr8HX8osV-ubuCJIrM9d7Xs6bJnmwTXbpO7pTJW","signatures":[{"protected":"eyJhbGciOiJFZERTQSIsImtpZCI6ImRpZDprZXk6ejZNa2oyUVQ4blpvaWF1WGNhMVA0SDZ6d0RqTjlSUDNORFdHazFBWFVkaGgyTjVwI3o2TWtqMlFUOG5ab2lhdVhjYTFQNEg2endEak45UlAzTkRXR2sxQVhVZGhoMk41cCJ9","signature":"78ikyfrUnOHtXXmSkeP6Y0nq9eEmV_Hw-tqNtWWsbIiDIMKdEgrjlZhlewlDjgrkBYTjvqtVSDqgufNA2NiYBA"}],"link":"bafyreib2dlevv7a5p4ulcx5on3qisivthv33l3hjwjtzwbgxn2j3xjjsky"},"linkedBlock":"omRkYXRhoWRzdGVwAGZoZWFkZXKiZnVuaXF1ZXA2VUM3YmxPQ2RWVW1VNUQva2NvbnRyb2xsZXJzgXg4ZGlkOmtleTp6Nk1rajJRVDhuWm9pYXVYY2ExUDRINnp3RGpOOVJQM05EV0drMUFYVWRoaDJONXA"} body.opts={"anchor":true,"publish":true,"sync":0,"syncTimeoutSeconds":0} http_version=1.1 req_header- status=200 content_length=379 content_type="application/json; charset=utf-8" ref=- user_agent="node-fetch/1.0 (+https://github.com/bitinn/node-fetch)" elapsed_ms=82.153 error_message="-" error_code=-
```

### CODE REVIEW

File: packages/cli/src/daemon/log-requests.ts
Comment (gvelez17) [inline packages/cli/src/daemon/log-requests.ts:45]: is this going to be extremely long?  agree we want it tho

**Review by gvelez17 — APPROVED** [review]
lgtm, only slightly worried about really long logs.  also assuming you tested the morgan.token code

**Comment by stbrody** [comment]
> also assuming you tested the morgan.token code

Not quite sure what you mean by this?  The testing I did was running the node, sending http requests to the node, and looking at the log messages generated.

File: packages/cli/src/daemon/log-requests.ts
Comment (stbrody) [reply packages/cli/src/daemon/log-requests.ts:45]: does seem like a possible risk, yes.  This is printing the full body of the commit being sent to js-ceramic, so is users are creating large json documents in their app, this will print the full document, which could be up to 256kb large

**Comment by gvelez17** [comment]
> > also assuming you tested the morgan.token code
> 
> Not quite sure what you mean by this? The testing I did was running the node, sending http requests to the node, and looking at the log messages generated.

yeah i meant if there were unit tests, its fine

Are we logging every request?  i'm still concerned we could fill up disks with this stuff - hopefully we could log this much detail only on error?

File: packages/cli/src/daemon/log-requests.ts
Comment (gvelez17) [reply packages/cli/src/daemon/log-requests.ts:45]: um if we do this for every single request on gitcoin i definitely think its a risk of crashing them by filling up the logs.  i would like to sample them

**Comment by stbrody** [comment]
yeah, agreed this is too verbose right now.  We should not merge this without some controls to avoid filling the log with literally every document written to Ceramic.

**Review by gvelez17 — CHANGES_REQUESTED** [review]
(we agreed that we need to make it a little lighter/more selective before merge)

do you want these maybe?  https://github.com/ceramicnetwork/js-ceramic/pull/3215

**Comment by stbrody** [comment]
This PR needs more work to be ready to merge, and since js-ceramic is dying soon and we've made it this long without this working, going to go ahead and close this unmerged

**Pull request closed without merging**
